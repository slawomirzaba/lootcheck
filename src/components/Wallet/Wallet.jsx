import React from 'react';
import { connect } from 'react-redux';
import { deposit, withdraw } from '../../reducers/actions/balanceActions';

export class Wallet extends React.Component {
    constructor(props) {
        super(props);
        this.state = { balanceInput: null };
    }

    onInputChange = (event) => this.setState({ balanceInput: parseFloat(event.target.value) });

    deposit = () => this.props.deposit(this.state.balanceInput);

    withdraw = () => this.props.withdraw(this.state.balanceInput);

    render() {
        return (
            <div>
                <h3 className='balance'>Wallet Deposit: {this.props.balance}</h3>
                <input className='balance-input' onChange={this.onInputChange} />
                <button className='btn-deposit' onClick={this.deposit}>Deposit</button>
                <button className='btn-withdraw' onClick={this.withdraw}>withdraw</button>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        balance: state.balance.balance
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        deposit: (value) => dispatch(deposit(value)),
        withdraw: (value) => dispatch(withdraw(value)),
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Wallet);