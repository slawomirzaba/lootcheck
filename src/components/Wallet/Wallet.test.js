import React from 'react';
import { shallow } from 'enzyme';
import { Wallet } from './Wallet';

describe('#Wallet', () => {
    const depositMock = jest.fn();
    const withdrawMock = jest.fn();
    const props = { balance: 20, deposit: depositMock, withdraw: withdrawMock };
    const wallet = shallow(<Wallet {...props} />);

    it('should correct render', () => {
        expect(wallet).toMatchSnapshot();
    })

    it('should correct display balance from props', () => {
        expect(wallet.find('.balance').text()).toEqual('Wallet Deposit: 20');
    });

    it('should render input', () => {
        expect(wallet.find('.balance-input').exists()).toBe(true);
    });

    describe('when user type in input', () => {
        const balanceInputValue = '12.2';

        beforeEach(() => {
            wallet.find('.balance-input').simulate('change', { target: { value: balanceInputValue } });
        });

        it('state should be changed and convert to int', () => {
            expect(wallet.state().balanceInput).toEqual(parseFloat(balanceInputValue));
        });

        describe('and user execute deposit action', () => {
            beforeEach(() => {
                wallet.find('.btn-deposit').simulate('click');
            });

            it('should be executed deposit action from props', () => {
                expect(depositMock).toBeCalledWith(parseFloat(balanceInputValue))
            });
        });

        describe('and user execute withdraw action', () => {
            beforeEach(() => {
                wallet.find('.btn-withdraw').simulate('click');
            });

            it('should be executed withdraw action from props', () => {
                expect(withdrawMock).toBeCalledWith(parseFloat(balanceInputValue))
            });
        });
    });

})