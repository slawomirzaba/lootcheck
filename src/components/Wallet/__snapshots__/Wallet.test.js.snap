// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Wallet should correct render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    balance={20}
    deposit={[Function]}
    withdraw={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="balance"
>
          Wallet Deposit: 
          20
</h3>,
        <input
          className="balance-input"
          onChange={[Function]}
/>,
        <button
          className="btn-deposit"
          onClick={[Function]}
>
          Deposit
</button>,
        <button
          className="btn-withdraw"
          onClick={[Function]}
>
          withdraw
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Wallet Deposit: ",
            20,
          ],
          "className": "balance",
        },
        "ref": null,
        "rendered": Array [
          "Wallet Deposit: ",
          20,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "balance-input",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Deposit",
          "className": "btn-deposit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Deposit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "withdraw",
          "className": "btn-withdraw",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "withdraw",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="balance"
>
            Wallet Deposit: 
            20
</h3>,
          <input
            className="balance-input"
            onChange={[Function]}
/>,
          <button
            className="btn-deposit"
            onClick={[Function]}
>
            Deposit
</button>,
          <button
            className="btn-withdraw"
            onClick={[Function]}
>
            withdraw
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Wallet Deposit: ",
              20,
            ],
            "className": "balance",
          },
          "ref": null,
          "rendered": Array [
            "Wallet Deposit: ",
            20,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "balance-input",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Deposit",
            "className": "btn-deposit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Deposit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "withdraw",
            "className": "btn-withdraw",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "withdraw",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
